astropy/wcs/wcs.py

```diff
@@ -3076,7 +3076,9 @@ def pixel_scale_matrix(self):
         except InconsistentAxisTypesError:
             try:
                 # for non-celestial axes, get_cdelt doesn't work
-                cdelt = np.dot(self.wcs.cd, np.diag(self.wcs.cdelt))
+                with warnings.catch_warnings():
+                    warnings.filterwarnings('ignore', 'cdelt will be ignored since cd is present', RuntimeWarning)
```

Comment:
do we really need to catch this warning here?

Reply:
Why not? It is it emitted by the test, either we fix it or choose to ignore it.

Reply:
This is not a test file. So if the warning is useful for production code, it shouldn't be catched. If it's not useful for production then we probably don't want to raise it anyway? 
(disclaimer: I don't know wcs internals, you may very well be right that this warning doesn't make sense at this exact spot, and thus should be ignored for everyone)

Reply:
Oh yeah, I missed that it is not a test file. I'll defer to subpackage maintainers then.

Reply:
This is deliberate, I don't think the user needs to see a warning in this case, especially since it's not something they can do anything about (in my view warnings should always be something the user can fix)